### PR TITLE
UML-2104 - New mock for data-lpa

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

Replace the python sirius mock from opg-data-lpa with a prism mock

Fixes UML-2104

## Approach

- (UML-2103) correct openapi spec for ual parameters and requests
- Replace data-lpa mocks with prism
- Add make target to update openapi spec in prism
- Add script to generate openapi-examples.yaml
- Map of parameters to responses in nginx.conf
- provide address to api-gateway service for api-app in docker-compose.yml
- Update call api gateway script to target a different dev endpoint
- Update call api gateway script to add logging and error handling
- point non-production environments to teh dev endpoint as the new persistant edpoint replacing integration. (this endpoint still targets the integration instance of Sirius)

## Learning
- https://mikefarah.gitbook.io/yq/operators
- https://meta.stoplight.io/docs/prism/ZG9jOjk1-http-mocking

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* ~New event_codes have been documented on the [wiki page](https://opgtransform.atlassian.net/wiki/spaces/LSML2/pages/3277881441/Understanding+the+event+logs)~
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* ~I have added tests to prove my work~
* ~I have added welsh translation tags and updated translation files~
* ~I have run an accessibility tool on any pages I have made changes to and fixed any issues found~
* ~The product team have tested these changes~

### CODE REVIEW

File: mock-integrations/merge.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+script_path=$(dirname $0)
+rm -f ${script_path}/swagger.yaml
+
```
Comment (andrewpearce-digital) [inline mock-integrations/merge.sh:5]: need to add a wget here to pull data-lpa's main branch version of the lpa-openapi.yml spec

File: docker-compose.dependencies.yml [hunk 2]
```diff
@@ -47,3 +47,23 @@ services:
     environment:
       REGION: eu-west-1
       SEED_PATH: /init/kms-seed.yaml
+
+  # ---------------------------
+  # Sirius gateway mock
+  api-gateway:
```
Comment (cooperaj) [inline docker-compose.dependencies.yml:53]: This should probably depend on data-lpa?

File: docker-compose.yml [hunk 3]
```diff
@@ -186,6 +186,8 @@ services:
     build:
       context: .
       dockerfile: service-api/docker/web/Dockerfile
+    # depends_on:
+    #   - api-app
```
Comment (cooperaj) [inline docker-compose.yml:190]: comments?

File: docker-compose.dependencies.yml (hunk 2 above)
Comment (andrewpearce-digital) [reply docker-compose.dependencies.yml:53]: thanks, done

File: mock-integrations/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# map specific LPA IDs to named examples in the Sirius Swagger file;
+# the returned_* examples all map to "Returned" status in the UI,
+# but use different response fields to derive the date
+map $request_uri $swagger_example {
+    # 200 examples
+    "~700000000047" 'lpa0047';
```
Comment (cooperaj) [inline mock-integrations/nginx.conf:6]: Might it be possible to generate this list using the list.txt lpas?

**Review by cooperaj — DISMISSED** [review]

File: mock-integrations/nginx.conf (hunk 4 above)
Comment (andrewpearce-digital) [reply mock-integrations/nginx.conf]: done. I'e generated the mappings and inserted them with sed as part of the make_examples script.

**Pull request merged**